keps/sig-cli/5091-kubectl-checkpoint/README.md

```diff
@@ -0,0 +1,1205 @@
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+As already mentioned in KEP-2008 one of the main risks creating a checkpoint
+is that all memory pages are written to disk. This includes all secrets like
+passwords, random numbers, private keys. The checkpoint archive that is
+written to the checkpoint directory of the kubelet is only accessible by
+the root user which already has the capability to access all that data.
+
+One approach to ensure that non of the secret information stored in
+the checkpoint archive is accessible would be to have encrypted checkpoint
+archives.  This is currently being developed in CRIU: [Add support for encrypted
+images][pr2297]
+
+In its first incarnation this KEP will not transfer the checkpoint archive
+from the node it was created on. The checkpoint archive will be created on
+the node the container is running on and the user will get the information
+about the node and the path to the checkpoint archive. This also means that
+nothing changes from a security point of view compared to KEP-2008.
+
+To ensure only users with appropriate access can create a checkpoint
+a new role needs to be introduced in the RBAC system. For the resource "pods"
+a new verb "checkpoint" will be introduced.
+
+TODO: Security review is still open.
+
+TODO: UX review is still open.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Currently the design details are based on the existing pull request: [Add
+'checkpoint' command to kubectl][pr120898]
+
+The API server is extended to handle checkpoint requests from *kubectl*:
+
+```go
+// PodCheckpointOptions is the query options to checkpoint a container
+type PodCheckpointOptions struct {
+    metav1.TypeMeta
+
+    // Container which to checkpoint
+    Container string
+
+    // timeout as specified in the kubelet API endpoint
+    timeout int64
+}
+```
+
+Also, *kubectl* is extended to call this new API server interface. The API
+server, upon receiving a request, will call the kubelet with the corresponding
+parameters passed from *kubectl*. Once the checkpoint has been successfully written
+to disk *kubectl* will return the name of the node as well as the location of
+the checkpoint archive to the user:
```

Comment:
as commented above, this is the most tricky part of the KEP, you need to expand on the technical design here, these endpoints are complex

Reply:
Unfortunately I am not sure what is needed here. I described the API to be just as the API provided by the `kubelet`. It just forwards everything 1:1 to the kubelet. Concerning different versions of api server and kubelet I described in one section that it will probably just return an error. I guess I do not get it what is required here.

Any existing examples I can take a look at?